pkg/providers/amifamily/ami.go

```diff
@@ -145,7 +150,7 @@ func (p *DefaultProvider) DescribeImageQueries(ctx context.Context, nodeClass *v
 }
 
 //nolint:gocyclo
-func (p *DefaultProvider) amis(ctx context.Context, queries []DescribeImageQuery) (AMIs, error) {
+func (p *DefaultProvider) amis(ctx context.Context, queries []DescribeImageQuery, includeDeprecated bool) (AMIs, error) {
```

Comment:
Why would we not always be including deprecated? What's the need for this to be an option on the interface? I would assume that every alias should just always include deprecated

Reply:
We could, but I wanted to limit the scope of the change to only affect the automatic upgrade use-case. I view pinned aliases as a shorthand for supplying AMI IDs and wanted the behavior to remain the same on both. Unless we want to backport deprecated AMI support altogether, I think drawing the line is the right move.

Reply:
Discussed offline, we're going to continue to limit the change to the automatic upgrade use-case.

Reply:
To add -- reason for this is only because it's a back-port -- our current state going forward is that we discover deprecated regardless of your selector type